src/surveys/components/RatingResponse.tsx

```diff
@@ -0,0 +1,62 @@
+import { Star } from 'lucide-preact';
+import { h } from 'preact';
+import { useEffect, useState } from 'preact/hooks';
+import { Header } from '../../components';
+import { ID, Question, RangeSettings, Theme } from '../../types';
+
+interface RatingResponseProps {
+  question: Question;
+  onAnswered: (questionId: ID, value: string) => void;
+  theme?: Theme;
+}
+
+function RatingResponse({ question, onAnswered, theme }: RatingResponseProps) {
+  const [answer, setAnswer] = useState(0);
+
+  useEffect(() => {
+    const answerQuestion = () => {
+      if (answer === 0) return;
+      onAnswered(question.id, answer.toString());
+    };
+
+    const timeoutId = setTimeout(answerQuestion, 500);
+
+    return () => {
+      clearTimeout(timeoutId);
+    };
+  }, [answer]);
+
+  return (
+    <div className={'px-6 space-y-3'}>
+      <Header
+        title={question.label}
+        description={question.description}
+        color={theme?.question}
+        centered
+      />
+
+      <div className='flex justify-center gap-6 px-6'>
+        {[1, 2, 3, 4, 5].map((num) => (
```

Comment:
@AfeezGL There are 2 issues discovered with this PR.

1. This PR assumes there's a static star counts. The length of `options` property in `Question` or `count` property in `RangeSettings` determines the star counts. If the `options` is empty, then use the `count` property. Either must be provided.
2. This PR also understandably assumes the icon shape will always be a `star`. The icon should be `star` `thumb` or `heart`. My suggestion is you add an optional `shape` to `RangeSettings`, if a value isn't provided then you can default to `star`.